### ISSUE TITLE
[General] Opt-in to Rust 2021 Edition with newest rustc

### ISSUE THREAD

**Issue by nfejzic** [unimarkup/unimarkup-rs#5]
Recently Rust 2021 Edition [was announced](https://blog.rust-lang.org/2021/10/21/Rust-1.56.0.html). 

This has some features which could be useful, i.e. [Disjoint closure captures](https://blog.rust-lang.org/2021/10/21/Rust-1.56.0.html#disjoint-capture-in-closures). 

However, using newest edition could break some edge cases. Also, if one writes a library using the newest edition, it limits the lowest supported `rustc` version. Since we're writing a binary, this should not be too big problem for us.
Also, since we're at an early stage of writing this project, we probably won't have many breaking cases, if any. 

Updating development environment is as simple as invoking `rustup update stable`. Adjusting project configuration should also be fairly trivial.

**TLDR:** Newest Rust Edition (2021) has some neat features. Could need some code adjusting, but this is highly unlikely at our development stage.
